src/configs/linkedin-passport.ts

```diff
@@ -0,0 +1,100 @@
+import type { Request } from 'express'
+import passport from 'passport'
+import { Strategy as JwtStrategy } from 'passport-jwt'
+import Profile from '../entities/profile.entity'
+import { dataSource } from './dbConfig'
+import {
+  JWT_SECRET,
+  LINKEDIN_CLIENT_ID,
+  LINKEDIN_CLIENT_SECRET,
+  LINKEDIN_REDIRECT_URL
+} from './envConfig'
+
+import { Strategy as LinkedInStrategy } from 'passport-linkedin-oauth2'
+import { findOrCreateUser } from '../services/auth.service'
+import { type CreateProfile, type User } from '../types'
+
+passport.use(
+  new LinkedInStrategy(
+    {
+      clientID: LINKEDIN_CLIENT_ID,
+      clientSecret: LINKEDIN_CLIENT_SECRET,
+      callbackURL: LINKEDIN_REDIRECT_URL,
+      scope: ['openid', 'email', 'profile'],
+      passReqToCallback: true
+    },
+    async function (
+      req: Request,
+      accessToken: string,
+      refreshToken: string,
+      profile: any,
```

Comment:
Can you specify the type here? @dsmabulage

Reply:
The method defined type is `passport.Profile` but in LinkedIn Login it returns a different type.

```
export interface LinkedInProfile {
  id: string
  givenName: string
  familyName: string
  picture: string
  email: string
}
```

So explicitly defined the type as `const data = profile as unknown as LinkedInProfile`